File: 练习/MyWeiBo/MyWeiBo/Class/Base/Controller/HJTabBarController.m
 //
//  HJTabBarController.m
//  MyWeiBo
//
//  Created by coco on 16/4/29.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "HJTabBarController.h"
#import "HJMyViewController.h"  //我的
#import "HJHomeViewController.h"  //主页
#import "HJMessageViewController.h"  //消息
#import "HJDiscoverViewController.h"  //发现
#import "HJNavigationController.h"
#import "HJTabBar.h"
#import "HJComposeController.h"  //发微博
@interface HJTabBarController ()

@end

@implementation HJTabBarController

- (void)viewDidLoad {
    [super viewDidLoad];
  
    [self addChildVC:[[HJHomeViewController alloc] init] title:NSLocalizedString(@"首页", nil)  image:@"tabbar_home" selectedImage:@"tabbar_home_selected"];
     [self addChildVC:[[HJMessageViewController alloc] init] title:NSLocalizedString(@"消息", nil)  image:@"tabbar_message_center" selectedImage:@"tabbar_message_center_selected"];
    [self addChildVC:[[HJDiscoverViewController alloc] init] title:NSLocalizedString(@"发现", nil) image:@"tabbar_discover" selectedImage:@"tabbar_discover_selected"];
    [self addChildVC:[[HJMyViewController alloc] init] title:NSLocalizedString(@"我的", nil) image:@"tabbar_profile" selectedImage:@"tabbar_profile_selected"];
    
    //利用KVC  使用自定义类
    HJTabBar *tabBar = [[HJTabBar alloc] init];
    [tabBar setClick:^{
        HJComposeController *Compose = [[HJComposeController alloc] init];
        HJNavigationController *navi = [[HJNavigationController alloc] initWithRootViewController:Compose];
        [self presentViewController:navi animated:YES completion:nil];
    }];
    [self setValue:tabBar forKeyPath:@"tabBar"];
    
    
}

/**
 *  @author XHJ, 16-04-29 15:04:04
 *
 *  设置控制器属性,添加到tabbarcontroller中
 *
 *  @param childVC       子控制器
 *  @param title         标题
 *  @param image         正常图片
 *  @param selectedImage 选中图片
 */
- (void)addChildVC:(UIViewController *)childVC
             title:(NSString *)title image:(NSString *)image
             selectedImage:(NSString *)selectedImage {
    /*
     //同下面的一句
    childVC.tabBarItem.title = title;
    childVC.navigationItem.title = title;
     */
    childVC.title = title;
    childVC.tabBarItem.image = [UIImage imageNamed:image];
    childVC.tabBarItem.selectedImage = [[UIImage imageNamed:selectedImage] imageWithRenderingMode:(UIImageRenderingModeAlwaysOriginal)]; //原始图片
    
    //设置tabbar文字样式
    [childVC.tabBarItem setTitleTextAttributes:@{NSForegroundColorAttributeName : UIColor.orangeColor} forState:(UIControlStateSelected)];
    [childVC.tabBarItem setTitleTextAttributes:@{NSForegroundColorAttributeName : UIColor.blackColor} forState:(UIControlStateNormal)];
    
    //添加导航栏
    HJNavigationController *navi = [[HJNavigationController alloc] initWithRootViewController:childVC];
    [self addChildViewController:navi];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: 小码哥大神5期练习/6、Swift版本-百思不得姐/Best Not Elder Sister/Best Not Elder Sister/Class/Base/Controller/HJTabBarController.swift
//
//  HJTabBarController.swift
//  Best Not Elder Sister
//
//  Created by coco on 16/7/22.
//  Copyright © 2016年 M-coppco. All rights reserved.
//

import UIKit

class HJTabBarController: UITabBarController {

    override class func initialize() {
        //统一设置,一般在控件添加之前设置,在它之前已经显示的控件,需要先remove再添加
        let tabBarItem = UITabBarItem.appearance()
        //文字
        tabBarItem.setTitleTextAttributes([NSForegroundColorAttributeName: UIColor.lightGrayColor()], forState: .Normal)
        tabBarItem.setTitleTextAttributes([NSForegroundColorAttributeName: UIColor.blackColor()], forState: .Selected)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        settingController(HJEssenceController(), title: "精华", image: "tabBar_essence_icon", selectImage: "tabBar_essence_click_icon")
        
        settingController(HJLastestController(), title: "最新", image: "tabBar_new_icon", selectImage: "tabBar_new_click_icon")
        
        settingController(HJConcernController(), title: "关注", image: "tabBar_friendTrends_icon", selectImage: "tabBar_friendTrends_click_icon")
        
        settingController(HJMyController(), title: "我", image: "tabBar_me_icon", selectImage: "tabBar_me_click_icon")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()

    }
    
    func settingController(controller: UIViewController, title: String, image: String, selectImage: String) {
        controller.title = title
        controller.tabBarItem.image = UIImage(named: image)
        //选择图片,默认选中的时候会有蓝色  方法1:原始图片(代码)  方法2:在Assets.xcassets中选中图片,  把Render as  选择original(Xcode)
        controller.tabBarItem.selectedImage = UIImage(named: selectImage)?.imageWithRenderingMode(.AlwaysOriginal)
        //文字
//        controller.tabBarItem.setTitleTextAttributes([NSForegroundColorAttributeName: UIColor.lightGrayColor()], forState: .Normal)
//        controller.tabBarItem.setTitleTextAttributes([NSForegroundColorAttributeName: UIColor.blackColor()], forState: .Selected)
        let navigationVC = HJNavigationController(rootViewController: controller)
        controller.view.backgroundColor = UIColor.hj_randomColor()
        self.addChildViewController(navigationVC)
    }

}
